Custom order filtering for ShipStation import XML

BillBushee opened the pull request:
I integrated this pull request https://github.com/onedesign/oneshipstation/pull/48 into my fork and then modified the order controller getOrders method to look for another plugin with a method named oneShipStationGetOrderCriteria. If the method exists, the returned ElementCriteriaModel will be used to find orders for the XML feed. If the user defined method is not found, then oneShipStation's default behavior won't change. I also updated the readme file to document the change.

BillBushee committed:
Update OneShipStation_XmlService.php

Increased custom fields code to truncate content at 1000 characters (limit of internal notes field length)

BillBushee committed:
Update OneShipStation_XmlService.php

Further revised code to truncate certain fields at 100 characters and the rest at 1000 characters

BillBushee committed:
Change code to get it back in sync with master

BillBushee committed:
Update OneShipStation_OrdersController.php

Added option to override criteria for getOrders

BillBushee committed:
Update README.md

Added paragraph and code example for overriding default getOrder search criteria

brianjhanson commented:
Hey @BillBushee,

This looks like a great addition to the plugin! It looks good to me at a glance, but we'll need to carve out some time to test it out a bit before pressing the big green button. I apologize in advance if this PR hangs out for awhile. 

Thanks again for the contribution!

BillBushee committed:
Update OneShipStation_XmlService.php

Fix bug in discount() method - was using number_format() resulting in a comma being added to float field.

BillBushee commented:
Hi @brianjhanson, 

I also discovered a little bug in the XML Service discount() method. It's using number_format() on the discount total. By default, that returns a comma as a number separator, so a discount of $1,000 or more breaks the import. I switched it to round() in my fork, since that's what the rest of the code appears to use. Would you like me to submit a separate pull request for that?

brianjhanson commented:
Hey @BillBushee, 

No need to open a new PR, especially since the change looks ver minor. Again, apologies this PR is kind of just hanging out. Hopefully over the long weekend I can carve out some time to get this merged in 🤞 

Thanks for your patience!
